components/collector/tests/source_collectors/dependency_track/test_source_up_to_dateness.py

```diff
@@ -4,42 +4,61 @@
 
 from dateutil.tz import tzlocal
 
-from source_collectors.dependency_track.base import DependencyTrackBase, DependencyTrackProject
+from source_collectors.dependency_track.base import (
+    DependencyTrackBase,
+    DependencyTrackMetrics,
+    DependencyTrackProject,
+)
 
-from tests.source_collectors.source_collector_test_case import SourceCollectorTestCase
+from .base_test import DependencyTrackTestCase
 
 
-class DependencyTrackSourceUpToDatenessVersionTest(SourceCollectorTestCase):
+class DependencyTrackSourceUpToDatenessVersionTest(DependencyTrackTestCase):
     """Unit tests for the Dependency-Track source up-to-dateness collector."""
 
     METRIC_ADDITION = "min"
     METRIC_TYPE = "source_up_to_dateness"
-    SOURCE_TYPE = "dependency_track"
     LANDING_URL = "https://dependency_track"
 
     def projects(self) -> list[DependencyTrackProject]:
         """Create Dependency-Track projects fixture."""
         now = datetime.now(tz=tzlocal()).replace(microsecond=0)
         self.yesterday = now - timedelta(days=1)
         yesterday_timestamp = round(self.yesterday.timestamp() * 1000)
+        self.day_before_yesterday = now - timedelta(days=2)
+        day_before_yesterday_timestamp = round(self.day_before_yesterday.timestamp() * 1000)
         self.last_week = now - timedelta(days=7)
-        last_week_timestamp = round(self.last_week.timestamp() * 1000)
+        self.last_week_timestamp = round(self.last_week.timestamp() * 1000)
         return [
-            DependencyTrackProject(name="Project 1", version="1.1", uuid="p1", lastBomImport=yesterday_timestamp),
-            DependencyTrackProject(name="Project 2", version="1.2", uuid="p2", lastBomImport=last_week_timestamp),
+            DependencyTrackProject(
+                name="Project 1",
+                version="1.1",
+                uuid="p1",
+                lastBomImport=yesterday_timestamp,
+                metrics=DependencyTrackMetrics(lastOccurrence=self.last_week_timestamp),
+            ),
+            DependencyTrackProject(
+                name="Project 2",
+                version="1.2",
+                uuid="p2",
+                lastBomImport=self.last_week_timestamp,
+                metrics=DependencyTrackMetrics(lastOccurrence=day_before_yesterday_timestamp),
+            ),
         ]
 
     def entities(self) -> list[dict[str, str]]:
         """Create the expected entities."""
         return [
             {
                 "key": "p1",
+                "last_bom_analysis": self.last_week.isoformat(),
```

Comment:
Perhaps this shouldn't be last week, but yesterday and `last_bom_import` day_before_yesterday, so the `test_filter_by_event_type_last_bom_analysis` and `test_filter_by_event_type_last_bom_import` don't both report `days=7`